VM/plot_vm.py

```diff
@@ -1,14 +1,50 @@
+"""
+Plots the VM domain and the SRF planes
+"""
+
 from argparse import ArgumentParser
 from logging import Logger
 from pathlib import Path
+
 from tempfile import TemporaryDirectory
 
 import numpy as np
+from shapely.geometry import Point, Polygon
 import yaml
 
 from qcore import geo, gmt, qclogging
 
 
+def is_SRF_inside_VM_domain(srf_corners: np.ndarray, vm_corners: str) -> bool:
+    """
+    Check if all SRF planes are inside the VM domain
+    Parameters
+    ----------
+    srf_corners : np.ndarray
+        Corners of SRF planes. Can be multiple planes,  Formatted as [plane1, plane2,...] where plane1=[[lon,lat],[lon,lat],[lon,lat],[lon,lat]]
+    vm_corners: str
+        Corners of the VM domain. Formatted as "lon\tlat\nlon\tlat\nlon\tlat\nlon\tlat\n"
+    """
+    vm_polygon = Polygon(
+        [
+            (float(lat_str), float(lon_str))
+            for lon_str, lat_str in [
+                point_str.split("\t")
+                for point_str in vm_corners.split("\n")
+                if len(point_str) > 0
+            ]
+        ]
+    )
+
+    all_inside = all(
+        vm_polygon.contains(Point(lat, lon))
+        for plane in srf_corners
+        for lon, lat in plane
+    )
```

Comment:
Got to be somewhat careful with this kind of code because shapely doesn't work with spherical coordinates and assumes everything lies on a flat plane. If you converted the coordinates to NZTM first, then it would work for sure.